src/rgw/rgw_kafka.cc

```diff
@@ -121,41 +151,26 @@ struct connection_t {
 
   // dtor also destroys the internals
   ~connection_t() {
-    destroy(status);
+    destroy();
   }
 };
 
-// convert int status to string - including RGW specific values
-std::string status_to_string(int s) {
-  switch (s) {
-    case STATUS_OK:
-        return "STATUS_OK";
-    case STATUS_CONNECTION_CLOSED:
-      return "RGW_KAFKA_STATUS_CONNECTION_CLOSED";
-    case STATUS_QUEUE_FULL:
-      return "RGW_KAFKA_STATUS_QUEUE_FULL";
-    case STATUS_MAX_INFLIGHT:
-      return "RGW_KAFKA_STATUS_MAX_INFLIGHT";
-    case STATUS_MANAGER_STOPPED:
-      return "RGW_KAFKA_STATUS_MANAGER_STOPPED";
-    case STATUS_CONF_ALLOC_FAILED:
-      return "RGW_KAFKA_STATUS_CONF_ALLOC_FAILED";
-    case STATUS_CONF_REPLCACE:
-      return "RGW_KAFKA_STATUS_CONF_REPLCACE";
-    case STATUS_CONNECTION_IDLE:
-      return "RGW_KAFKA_STATUS_CONNECTION_IDLE";
-  }
-  return std::string(rd_kafka_err2str((rd_kafka_resp_err_t)s));
-}
-
 void message_callback(rd_kafka_t* rk, const rd_kafka_message_t* rkmessage, void* opaque) {
   ceph_assert(opaque);
 
   const auto conn = reinterpret_cast<connection_t*>(opaque);
   const auto result = rkmessage->err;
 
+  if (rkmessage->err == 0) {
+      ldout(conn->cct, 20) << "Kafka run: ack received with result=" << 
+        rd_kafka_err2str(result) << dendl;
+  } else {
+      ldout(conn->cct, 1) << "Kafka run: nack received with result=" << 
+        rd_kafka_err2str(result) << dendl;
```

Comment:
nit, if you could add the broker and topic name here and also for error after calling `rd_kafka_produce` at line 428
"topic: " << rd_kafka_name(rk) << " broker: " << conn->broker << dendl

Reply:
will do a round of debug log fixes (in a different PR), and add that there.
* use DoutPrefixProvider
* unify what log messages contain (broker+topic)